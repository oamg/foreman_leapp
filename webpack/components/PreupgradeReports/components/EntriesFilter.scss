.entries-filter-reset-height {
  height: inherit;
}
