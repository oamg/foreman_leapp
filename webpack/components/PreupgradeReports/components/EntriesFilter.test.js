import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import EntriesFilter from './EntriesFilter';

const onFilterValueChange = () => {};
const onFilterTypeChange = () => {};

const fixtures = {
  'should render for title filter': {
    onFilterTypeChange,
    onFilterValueChange,
    filterType: 'title',
    filterValue: 'version',
  },
  'should render for severity filter': {
    onFilterTypeChange,
    onFilterValueChange,
    filterType: 'severity',
    filterValue: 'low',
  },
  'should render for host filter': {
    onFilterTypeChange,
    onFilterValueChange,
    filterType: 'hostname',
    filterValue: 'foo',
  },
};

describe('EntriresFilter', () =>
  testComponentSnapshotsWithFixtures(EntriesFilter, fixtures));
