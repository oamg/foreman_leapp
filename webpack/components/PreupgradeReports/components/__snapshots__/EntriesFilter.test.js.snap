// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntriresFilter should render for host filter 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={4}
  >
    <FormGroup
      bsClass="form-group"
    >
      <InputGroup
        bsClass="input-group"
      >
        <DropdownButton
          componentClass={[Function]}
          id="entry-filter"
          title="Host"
        >
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="title"
            onClick={[Function]}
          >
            Title
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="severity"
            onClick={[Function]}
          >
            Risk Factor
          </MenuItem>
          <MenuItem
            active={true}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="hostname"
            onClick={[Function]}
          >
            Host
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="fix"
            onClick={[Function]}
          >
            Fix Type
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="inhibitor"
            onClick={[Function]}
          >
            Inhibitor
          </MenuItem>
        </DropdownButton>
        <DebounceInput
          className="form-control entries-filter-reset-height"
          debounceTimeout={300}
          element="input"
          forceNotifyByEnter={true}
          forceNotifyOnBlur={true}
          minLength={0}
          onChange={[Function]}
          type="text"
        />
      </InputGroup>
    </FormGroup>
  </Col>
</Row>
`;

exports[`EntriresFilter should render for severity filter 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={4}
  >
    <FormGroup
      bsClass="form-group"
    >
      <InputGroup
        bsClass="input-group"
      >
        <DropdownButton
          componentClass={[Function]}
          id="entry-filter"
          title="Risk Factor"
        >
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="title"
            onClick={[Function]}
          >
            Title
          </MenuItem>
          <MenuItem
            active={true}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="severity"
            onClick={[Function]}
          >
            Risk Factor
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="hostname"
            onClick={[Function]}
          >
            Host
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="fix"
            onClick={[Function]}
          >
            Fix Type
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="inhibitor"
            onClick={[Function]}
          >
            Inhibitor
          </MenuItem>
        </DropdownButton>
        <DropdownButton
          id="entry-value-filter"
          title="Low"
        >
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key=""
            onClick={[Function]}
          >
            All
          </MenuItem>
          <MenuItem
            active={true}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="low"
            onClick={[Function]}
          >
            Low
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="medium"
            onClick={[Function]}
          >
            Medium
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="high"
            onClick={[Function]}
          >
            High
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="info"
            onClick={[Function]}
          >
            Info
          </MenuItem>
        </DropdownButton>
      </InputGroup>
    </FormGroup>
  </Col>
</Row>
`;

exports[`EntriresFilter should render for title filter 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={4}
  >
    <FormGroup
      bsClass="form-group"
    >
      <InputGroup
        bsClass="input-group"
      >
        <DropdownButton
          componentClass={[Function]}
          id="entry-filter"
          title="Title"
        >
          <MenuItem
            active={true}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="title"
            onClick={[Function]}
          >
            Title
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="severity"
            onClick={[Function]}
          >
            Risk Factor
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="hostname"
            onClick={[Function]}
          >
            Host
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="fix"
            onClick={[Function]}
          >
            Fix Type
          </MenuItem>
          <MenuItem
            active={false}
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
            key="inhibitor"
            onClick={[Function]}
          >
            Inhibitor
          </MenuItem>
        </DropdownButton>
        <DebounceInput
          className="form-control entries-filter-reset-height"
          debounceTimeout={300}
          element="input"
          forceNotifyByEnter={true}
          forceNotifyOnBlur={true}
          minLength={0}
          onChange={[Function]}
          type="text"
        />
      </InputGroup>
    </FormGroup>
  </Col>
</Row>
`;
